### ISSUE TITLE
RFC: separate docs into "docs" and "guides"

### ISSUE THREAD

**Issue by dwalkr** [tinacms/tinacms.org#338]
Our documentation needs to serve a spectrum of users, ranging from those looking for a quick + easy way to get started with Tina, to those interested in wielding parts of the API on their own for highly custom use cases, or building integrations. For simplicity I'll refer to these as **lower-investment** and **higher-investment** groups, referring to how much time/energy they want to invest in understanding/implementing Tina.

Trying to serve this entire spectrum from the documentation is challenging and can lead to the docs being confusing/unhelpful.

## Proposal

1. Add a new top-level nav item to tinacms.org, titled "Guides"
2. Remove the Tina+Gatsby and Tina+NextJS sections from /docs and repurpose them into Guides as appropriate

### Docs

Docs should focus on the APIs and components of Tina, and cater primarily to developers in the **medium- and higher-investment** part of the spectrum. Naive examples to demonstrate functionality (like what's in the Plugin docs) would be preferred over examples that assume a specific setup.

We should consider a linear path through the documentation. Docs should attempt to present the most common/ubiquitous APIs and components first, leaving more esoteric or uncommonly-needed features for later. This organization should not come at the expense of cohesiveness, however.

### Guides

Guides aim to serve the **lower- to medium-investment** developers, but will likely be useful to everyone.

The attitude of a Guide will be somewhere between documentation and a blog post. Guides should start with an overview of what will be achieved, and may be broken up into multiple logical steps, each in its own markdown file. 

```
| guides
\
  |--nextjs-github-public-repo
  \
    |--overview.md
    |--1-add-github-api.md
    |--2-bootstrap-api-routes.md
```

We should encourage guides to be as small in scope as seems appropriate, and have guides reference each other. For example, a guide on using the GitHub packages with NextJS should assume you have already bootstrapped Tina into your NextJS site (adding `TinaProvider` to `_app.js`,) and should link to a guide on how to do that rather than explaining it all over again.

### Discoverability

Guides should be:
1. searchable via algolia
2. discoverable via tags

#### Guide Layout

something like this

![image](https://user-images.githubusercontent.com/15221702/79360172-8887e380-7f11-11ea-967d-72243dfd1135.png)

**Comment by kendallstrautman** [comment]
Here are some of my thoughts after trying to get a better grasp on this issue and referencing how other projects attempt to solve these problems. I don't think we should necessarily implement everything I suggested now, but this is more of a long-term assessment on how we could improve our documentation.

## Re: The documentation problem

### Different types of folks who might be using Tina:

- **Explorers:** who want to get up and running quickly, don’t mind using a pre-baked demo to see what Tina is all about and how it works. They need quickstarts, tutorials, fun examples, easy-to-understand helpers, more magic, they need to get from 0 to editing quickly.

- **Implementers:** who want to actually _use_ Tina on their project. They don’t want to use a starter, but need enough information to bootstrap or manually add Tina to their project. They need generalized examples, with enough specificity to guide them through any roadblocks they may hit in setup. They need to have a basic understanding of the Tina building blocks and what’s involved with editing a site with Tina from 0 to prod.

- **Contributors:** who want to understand how Tina works from a deeper-level. These folks may be creating plugins, setting up alternative data storage solutions, or contributing new ideas or features to the codebase. They need detailed API documentation and advanced conceptual documentation.

**Our docs need to accommodate all these folks**, providing clear paths for them to follow based on their engagement-level and intent. 

Some feedback we’ve gotten is that the _docs are too prescriptive_, or that they seem _directed at an ‘advanced’ developer._ This makes sense when you look at the Gatsby docs where they’re literally showing people how to install Xcode — command-line 101 type stuff. 

Our docs definitely assume a certain level of competency and comfortability in working with OS js projects. We _assume_ you know Gatsby. We _assume_ you know Next.js (depending on what you’re using). We _assume_ you have a package manager installed and are comfortable with API terminology, Typescript etc. 

I don’t necessarily think it’s a bad thing that our docs come across as more advanced. The reality is that using Tina is not a simple 2 step setup process and for where the project is at, a little more perseverance is required to be successful. 

### What can be learned from other orgs and projects

- I think it is most interesting to reference documentation for products that interact with a lot of technology — [Stripe](https://stripe.com/docs) & [Shopify](https://www.shopify.dev/concepts/shopify-introduction) are good examples of broad but well organized docs. 

- I also thought [Prismic](https://prismic.io/docs) did a good job of providing easy ways to see specific information based on the framework or language you are using. They have icons for the language or framework that link to specific docs and guides. Similar to what we discussed yesterday of a ‘choose your own adventure’ type app for linking folks to corresponding info.

- [Gatsby docs](https://www.gatsbyjs.org/docs/) are turning into an encyclopedia. They even have a [glossary](https://www.gatsbyjs.org/docs/glossary/), which I think is cool, but not sure how useful it really is. They have a _ton_ of docs — maybe too much to sort through. But I think they do a good job of guiding folks towards differing levels of hand-holding with [quickstart](https://www.gatsbyjs.org/docs/quick-start/) + [recipes](https://www.gatsbyjs.org/docs/recipes/) + [guides](https://www.gatsbyjs.org/docs/guides/) + [tutorials](https://www.gatsbyjs.org/tutorial/). Their docs menu has 3 levels of nesting. Tutorials & contributing guides have their own dedicated pages. I think they do a great job of accommodating entry-level developers who want to build a site quickly.

- [Next.js docs](https://nextjs.org/docs/getting-started) are more sparse but somehow they work. The ‘learning’ intro game is a neat way to get folks on the right track. And their simple `creat-next-app` is a cool starting point (is it too simple though?). Their docs also have an ‘Advanced Features’ tab that hides complexity from first-time folks. 

## Ideas for improvement

- The Tina docs intro page could have more information on the way people might use Tina ([see shopify](https://www.shopify.dev/concepts/ways-to-build)). This intro should also be the point at which we send folks down different paths — quickstart, generic setup, advanced concepts etc. 

- I think we could reintroduce a ‘concepts’ tab and bury some of the more advanced topics in there (plugins, APIs, etc.). 

- Do we call this new tab _Guides_ or _Tutorials_? This is how Gatsby differentiates the two: “While the tutorials are step-by-step instructions, reference guides are resources about the various Gatsby development techniques.” If we go with guides, we could use the blog to make ‘tutorials’

- I think having generic quickstarts for the frameworks we support is important. 

- We need to think about the baseline information for using Tina that is common across all frameworks: What is a form, creating a form. What is a field, how to configure fields etc. What about data storage? What are the UI components for editing (sidebar versus inline etc)? 

- **We need to do a better job of explaining _What Tina Is_** — I think it could be a cool _thought experiment_ for us all individually to try and explain Tina in 500 words or less (or 5 minutes if you’re a talker), and see what we come up with. Some questions to get the thoughts churning might be: What is TinaCMS? What Frameworks does it work with? What about Data Storage? Give an story of someone who would use TinaCMS.

- Some uber bare-bones starters for the different frameworks (maybe without data storage set up at all) could be really useful and easier to maintain for lack of complexity. (a la `creat-next-app`).

- There should be more cross-linking between the blog and the docs. The blogs could still serve as our place to hash out ideas or provide fun tutorials. We need to make sure that these blogs are linked to and coming up in the docs if relevant.

- Those generated API docs would be clutch. 

## Thoughts on a new docs structure

I don’t think Guides should exist in the global tinacms.org nav, but should rather be added right below the Getting Started tab. This tab item could link to a new page (same docs layout) to have it’s own nav. Reasons being: 
- Aesthetically it won’t look good to add another bubble button (lol not huge priority but I gotta say it)
- As a dev, when I’m learning something new or looking for an answer I instinctively search for docs
- I think having a single entry point is better for now, seeing as there isn’t a huge amount of docs-menu nav items and pages etc.

```
- Getting-started
    - Intro
    - How Tina works - We should rewrite this with what we learn from trying to explain what tina is in the thought experiment described above.
- Guides (or Tutorials) — This could link to a Guides page with the same layout but a separate nav list
    - Next.js 
      - Next.js quickstart
      - Next.js with github
    - Gatsby
      - Gatsby quickstart
      - Gatsby with Git
      - Gatsby Cloud
- Forms
- Fields
- Media
- Styles
- Data Storage
  - Overview
  - Git
  - Github
- Inline Editing
  - Overview
  - Inline Blocks
- Concepts - these should be written with _contributors_ in mind.
  - The CMS
  - Plugins
  - Alerts
- Contributing
```

**Comment by dwalkr** [comment]
> Aesthetically it won’t look good to add another bubble button (lol not huge priority but I gotta say it)

design shouldn't drive content; if it were to become a problem we should revisit the design

---

> As a dev, when I’m learning something new or looking for an answer I instinctively search for docs

I agree with this. I think it would be appropriate to have something in the docs introduction to funnel users looking for an end-to-end tutorial toward Guides. Like you suggested here:

> The Tina docs intro page could have more information on the way people might use Tina (see shopify). This intro should also be the point at which we send folks down different paths — quickstart, generic setup, advanced concepts etc.

---

> I think having a single entry point is better for now, seeing as there isn’t a huge amount of docs-menu nav items and pages etc.

I agree about a single entry point, but IMO the leaner we can make the docs, the better. Over time they _will_ become huge, it's the nature of the beast

> Next.js docs are more sparse but somehow they work. The ‘learning’ intro game is a neat way to get folks on the right track

The small size of Next.js' docs is a huge advantage to me. This is a little bit personal preference, but it makes Next feel like a simple tool vs. Gatsby's complicated machine. My inspiration for Guides here is mainly influenced by their approach.

---

> Concepts - these should be written with _contributors_ in mind

I disagree here. The docs on tinacms.org should be geared toward users/integrators; contributors should look to github READMEs and the generated API docs. This was discussed internally and is probably our fault for not communicating this publicly.

**Comment by kendallstrautman** [comment]
> design shouldn't drive content 

 Fair enough haha yeah I see how it makes sense for _Guides_ to be separated out now with the long term in mind 👌🏼 

> The small size of Next.js' docs is a huge advantage to me.

I think it's a delicate balance, but I agree they provide a great, streamlined reference.

>  Concepts - these should be written with contributors in mind

I think what I really meant here was to 'folks who want to dive deep' (referencing the 3rd persona type outlined above, bad name choice 🤦‍♀️) , but that's a good point regarding the README's and generated API docs — I didn't realize that.

**Comment by dwalkr** [comment]
I see "Guides" being its own area of the site eventually, because I envision a large number of guides. For now we will probably only have a handful, so maybe we can start it out inside the docs nav and move it out when it becomes unwieldy.

**Comment by kendallstrautman** [comment]
Will be implemented by #374
